TeamCode/src/main/java/org/firstinspires/ftc/teamcode/PixyCam.java

```diff
@@ -0,0 +1,170 @@
+    private final byte[] PIXY_DEVICE_SIGNATURE = {85, (byte) 170, 85, (byte) 170};
+    private final int SIGNATURE_SPACE_SIZE = 10; // Device signature will be in the first 10 bytes
+
+    PixyCam() {}
+    PixyCam( I2cDeviceSynch deviceClient ) { initialize( deviceClient); }
+
+    public void initialize( I2cDeviceSynch deviceClient )
+    {
+        m_deviceClient = deviceClient;
+        m_deviceClient.engage();
+    }
+
+    public byte[] GetRegisterValues() { return m_registerValues; }
+
+    /**
+     * Update the cashed register values
+     * @return whether the operation was successful
+     */
+    public boolean Update()
+    {
+
+        byte[] registerValues = m_deviceClient.read( START_ADDRESS, READ_LENGTH );
+        byte[] signatureSpace = new byte[SIGNATURE_SPACE_SIZE];
+
+        // Only use the first ten bytes for signature scanning
+        for( int iIndex = 0; iIndex < SIGNATURE_SPACE_SIZE; iIndex++)
+        {
+            signatureSpace[iIndex] = (byte) mod(registerValues[iIndex], 256);
+        }
+
+        // If all signature space values are zero then we are not retrieving valid data.
+        int signatureIndex = indexOfSubArray(signatureSpace, PIXY_DEVICE_SIGNATURE);
+        if( signatureIndex > -1 )
+        {
+            m_registerValues = registerValues;
+            m_dataStartAddress = signatureIndex;
+            return true;
+        }
+        else
+            return false;
+    }
+
+    /**
+     * Update the register values. If the values are not available, halt the thread until they are.
+     */
+    public void ForceUpdate()
+    {
+        while(!Update());
+    }
+
+    public int GetDataStartAddress()
+    {
+        return m_dataStartAddress;
+    }
+
+    /**
+     * @return the X coordinate of the center of the found object. Returns -1 if no data is
+     * available from the pixy cam
+     */
+    public int GetXCenter() { return ReadSignedInt(PixyDataOffsets.X_CENTER);}
+
+    /**
+     * @return the Y coordinate of the center of the found object. Returns -1 if no data is
+     * available from the pixy cam
+     */
+    public int GetYCenter() { return ReadSignedInt(PixyDataOffsets.Y_CENTER); }
+
+    /**
+     * @return the width of the found object. Returns -1 if no data is available from the pixy cam.
+     */
+    public int GetWidth() { return ReadSignedInt(PixyDataOffsets.WIDTH); }
+
+    /**
+     * @return the height of the found object. Returns -1 if no data is available from the pixy cam.
+     */
+    public int GetHeight() { return ReadSignedInt(PixyDataOffsets.HEIGHT); }
+
+    /**
+     * Java % operator is remainder not mod
+     * @return result of modulus operation
+     */
+    private int mod(int x, int y)
+    {
+        int result = x % y;
+        return result < 0? result + y : result;
+    }
+
+    /**
+     * Given array and subArray, determines if subArray is in array
+     * @return index of the subArray in array. Returns -1 if not present
+     */
```

Comment:
Same comment about reusing standard code whenever possible. If you can use Java 7+, you can probably use [`Collections.indexOfSubList()`](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#indexOfSubList(java.util.List,%20java.util.List)).